Sources/Remote Logging/Crash Logging/CrashLogging.swift

```diff
@@ -168,54 +171,24 @@ public extension CrashLogging {
     }
 
     /// Sends an `Event` to Sentry and triggers a callback on completion
-    func logErrorImmediately(_ error: Error, userInfo: [String: Any]? = nil, level: SentryLevel = .error, callback: @escaping (Result<Bool, Error>) -> Void) throws {
+    func logErrorImmediately(_ error: Error, userInfo: [String: Any]? = nil, level: SentryLevel = .error, callback: @escaping () -> Void) throws {
         try logErrorsImmediately([error], userInfo: userInfo, level: level, callback: callback)
     }
 
-    func logErrorsImmediately(_ errors: [Error], userInfo: [String: Any]? = nil, level: SentryLevel = .error, callback: @escaping (Result<Bool, Error>) -> Void) throws {
-
-        var serializer = SentryEventSerializer(dsn: dataProvider.sentryDSN)
+    func logErrorsImmediately(_ errors: [Error], userInfo: [String: Any]? = nil, level: SentryLevel = .error, callback: @escaping () -> Void) throws {
 
         errors.forEach { error in
-            let event = Event.from(
-                error: error as NSError,
-                level: level,
-                user: dataProvider.currentUser?.sentryUser,
-                extra: userInfo ?? (error as NSError).userInfo
-            )
-
-            event.threads = currentThreads()
-
-            serializer.add(event: event)
-        }
-
-        guard let requestBody = try? serializer.serialize() else {
-            TracksLogError("⛔️ Unable to send errors to Sentry – error could not be serialized. Attempting to schedule delivery for another time.")
-            errors.forEach {
-                SentrySDK.capture(error: $0)
+            // Amending the global scope on a per-event basis seems like the best way to add the
+            // caller-provided `userInfo` and `level`.
+            SentrySDK.capture(error: error) { scope in
+                // Under the hood, `setExtras` uses `NSMutableDictionary` `addEntriesFromDictionary`
+                // meaning this won't replace the whole extras dictionary.
+                scope.setExtras(userInfo)
+                scope.setLevel(level)
             }
-            return
-        }
-
-        let dsn = try SentryDsn(string: dataProvider.sentryDSN)
-        guard let authString = dsn.getAuthString() else {
-            throw Errors.unableToConstructAuthStringError
         }
-
-        var request = URLRequest(url: dsn.getEnvelopeEndpoint())
-        request.httpMethod = "POST"
-        request.httpBody = requestBody
-        request.addValue(authString, forHTTPHeaderField: "X-Sentry-Auth")
-
-        URLSession.shared.dataTask(with: request) { (responseData, urlResponse, error) in
-            if let error = error {
-                callback(.failure(error))
-                return
-            }
-
-            let didSucceed = 200...299 ~= (urlResponse as! HTTPURLResponse).statusCode
-            callback(.success(didSucceed))
-        }.resume()
+        SentrySDK.flush(timeout: CrashLogging.eventsFlushingTimeout)
```

Comment:
Putting this into a background queue so that we don't block `logErrorsImmediately` callers?

Reply:
Or adding a `wait: Bool` argument, which'd make the `logErrorAndWait` simpler to implement.